src/vs/workbench/electron-browser/shell.ts

```diff
@@ -291,6 +291,35 @@ export class WorkbenchShell {
 		});
 	}
 
+	// {{SQL CARBON EDIT}}
+	private sendUsageEvents(): void {
+		const dailyLastUseDate = Date.parse(this.storageService.get('telemetry.dailyLastUseDate'));
+		const weeklyLastUseDate = Date.parse(this.storageService.get('telemetry.weeklyLastUseDate'));
+		const monthlyLastUseDate = Date.parse(this.storageService.get('telemetry.monthlyLastUseDate'));
+
+		let today = new Date().toUTCString();
+
+		// daily user event
+		if (this.diffInDays(Date.parse(today), dailyLastUseDate) > 1) {
```

Comment:
should all the comparisons be `>=`?  I think that `tomorrow-today==1` and we'd want to send the event?